readerwriterqueue.h

```diff
@@ -40,6 +40,10 @@
 #endif
 #endif
 
+#if !defined _MSC_VER || _MSC_VER >= 1800 // variadic templates: either a non-MS compiler or VS >= 2013
```

Comment:
Can you add the parentheses around `defined` for consistency?
Also, this whole block should be wrapped with an `#ifndef MOODYCAMEL_HAS_EMPLACE` in case other headers want to define it as well.

Reply:
Done.